src/neo/Network/P2P/RemoteNode.cs

```diff
@@ -227,6 +232,10 @@ private void OnVersionPayload(VersionPayload version)
                 return;
             }
             SendMessage(Message.Create(MessageCommand.Verack));
+            if (IsFullNode)
+            {
+                SendMessage(Message.Create(MessageCommand.Mempool));
```

Comment:
Doesn't it break the handshake in that you're sending some regular message before receiving a `Verack` from the other side?

Reply:
I think we should sync the blocks first.

Reply:
yes, maybe we should wait for verack

Reply:
true